src/main/java/de/tum/cit/aet/artemis/core/security/jwt/TokenProvider.java

```diff
@@ -95,11 +95,26 @@ public long getTokenValidity(boolean rememberMe) {
      * @return JWT Token
      */
     public String createToken(Authentication authentication, boolean rememberMe) {
+        return createToken(authentication, getTokenValidity(rememberMe));
+    }
+
+    /**
+     * Create JWT Token a fully populated <code>Authentication</code> object.
+     *
+     * @param authentication Authentication Object
+     * @param duration       the Token lifetime
+     * @param tools          tools this token is used for
+     * @return JWT Token
+     */
+    public String createToken(Authentication authentication, long duration, String... tools) {
         String authorities = authentication.getAuthorities().stream().map(GrantedAuthority::getAuthority).collect(Collectors.joining(","));
 
+        String toolClaims = String.join(",", tools);
+
         long now = (new Date()).getTime();
```

Comment:
I agree. I think it might actually lead to problems if we do *not* use ZonedDateTime here. Unless the jwtBuilder has some issues with using ZonedDateTime, I don't see a reason to not use it.

Reply:
use of System.currentTimeMillis()should solve the issue